exec/java-exec/src/main/java/org/apache/drill/metastore/expr/FilterPredicate.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.metastore.expr;
+
+import org.apache.drill.exec.expr.stat.RowsMatch;
+
+public interface FilterPredicate<T extends Comparable<T>> {
```

Comment:
When creating these interfaces regarding Filters, let's think of them being shared between both planning time and run-time because we want to allow for run-time pruning for parquet row groups.  During run-time, I would think there is no reason to go through the Metastore.  

cc @Ben-Zvi

Reply:
Agree that this interface may be used at run-time, will move it to another package. Usage of this interface assumes using `ColumnStatistics` classes which may be populated by the statistics from table footers.

Reply:
Should this be made independent of metastore ?